### ISSUE TITLE
axes limits revert to automatic on sharing axes?

### ISSUE THREAD

**Issue by gregcolyer** [matplotlib/matplotlib#10062]
Sharing axes undoes manual limits, but they can be restored afterwards.

The script below initially has manual limits on the first (and only) subplot. After uncommenting just the lines below 1, the manual limits are lost (on both subplots). After uncommenting the lines below 2, they are restored (on both subplots). It may be by design, but is a gotcha for users because usually (without axis sharing) the limits are not sensitive to where they are set (e.g. before or after calling `contour`). The sensitivity to placement with shared axes does not seem to be easily found in the documentation.

matplotlib 2.0.0 via conda

```python
import matplotlib.pyplot as plt
import numpy as np

p = np.arange(100,1000,100)
l = np.arange(-80,90,40)
t = np.random.random_sample((p.size,l.size))

plt.figure()

ax1 = plt.subplot(221)
ax1.set_xlim(-90,90)    ## Set manual limits
ax1.set_ylim(1000,0)    ##
c1 = ax1.contour(l,p,t)

# 1. Adding a second subplot reverts to automatic limits on both subplots ...
#ax2 = plt.subplot(223,sharex=ax1,sharey=ax1)
#c2 = ax2.contour(l,p,t)

# 2. ... unless set manual limits after
#ax1.set_xlim(-90,90)
#ax1.set_ylim(1000,0)

plt.show()
```

**Comment by jklymak** [comment]
Its possible that the logic in `cla` could be changed to let this happen.  Right now `auto` gets set to `None`, but you probably want it set to `self._sharex.get_autoxscale()`   I'm not sure what the knock-on effects might be of that change however.  

https://github.com/matplotlib/matplotlib/blob/fb2c15b361cd5de5121d16699e7609376ccae72a/lib/matplotlib/axes/_base.py#L1004-L1012

**Comment by lkjell** [comment]
It is not sharing axes that breaks the datalim it is the contourplot. Can check with 
`
print(ax1.get_xlim())`

Anyway in def contour you have `self.autoscale_view()` called.

**Comment by jklymak** [comment]
I believe that `ax1.autoscale_view()` doesn't do anything if `set_xlim` has been called because `ax1._autoscaleXon` is False.  The problem is that when ax2 gets added as a twin, it has `self._autoscaleXon` of True, the axes limits are reset fo ax2, but its twinned, so ax1 also gets reset to the auto limits.  The OP is arguing that if `ax1._autoscaleXon` is False, it should be False for `ax2` as well.

**Comment by phobson** [comment]
closed via #10798

**Comment by gregcolyer** [comment]
@shaynair Thank-you for addressing this
